cli/src/cli_util.rs

```diff
@@ -2278,6 +2278,44 @@ pub fn run_ui_editor(settings: &UserSettings, edit_path: &PathBuf) -> Result<(),
     Ok(())
 }
 
+pub fn edit_temp_file(
+    error_name: &str,
+    tempfile_suffix: &str,
+    repo_path: &Path,
+    content: &str,
+    settings: &UserSettings,
+) -> Result<String, CommandError> {
+    let file = (|| -> Result<_, io::Error> {
+        let mut file = tempfile::Builder::new()
+            .prefix("editor-")
+            .suffix(tempfile_suffix)
+            .tempfile_in(repo_path)?;
+        file.write_all(content.as_bytes())?;
+        file.seek(SeekFrom::Start(0))?;
+        Ok(file)
+    })()
+    .map_err(|e| {
+        user_error(format!(
+            "Failed to create {} file in \"{}\": {}",
+            error_name,
+            repo_path.display(),
+            e
+        ))
+    })?;
+
+    let file_path = file.path().to_owned();
+    run_ui_editor(settings, &file_path)?;
+
+    let mut buf = String::new();
+    io::BufReader::new(file).read_to_string(&mut buf)?;
```

Comment:
These `seek()`, keeping `file` opened, and `io::BufReader` no longer make sense. It's probably better to based off `edit_description()`.

FWIW, the original `edit_sparse()` looked scary. It wouldn't work if the editor created new file instead of truncating.

Reply:
Done.